packages/next/src/server/async-storage/static-generation-async-storage-wrapper.ts

```diff
@@ -63,6 +63,13 @@ export const StaticGenerationAsyncStorageWrapper: AsyncStorageWrapper<
       !renderOpts.isDraftMode &&
       !renderOpts.isServerAction
 
+    const prerenderState =
```

Comment:
Nit: To make it more explicit, using something like:

```suggestion
    const prerenderState: StaticGenerationStore['prerenderState'] =
```

Ensures that the type errors are isolated to the point at which they're created rather than delegating further down.